token_exchange/plugin.py

```diff
@@ -0,0 +1,10 @@
+from openforms.pre_requests.base import PreRequestHookBase
+from openforms.pre_requests.registry import register
+
+from .auth import TokenAccessAuth
+
+
+@register("token_exchange")
+class TokenExchangePreRequestHook(PreRequestHookBase):
+    def __call__(self, method: str, url: str, **kwargs):
+        kwargs["auth"] = TokenAccessAuth
```

Comment:
```suggestion
        kwargs["auth"] = TokenAccessAuth()
```
maybe also put a check that there's no `auth` yet in the kwargs, and if there is, emit a `logger.warning` that we're overwriting it